xds/internal/balancer/edsbalancer/eds.go

```diff
@@ -189,17 +184,12 @@ func (x *edsBalancer) handleGRPCUpdate(update interface{}) {
 	}
 }
 
-func (x *edsBalancer) handleXDSClientUpdate(update interface{}) {
-	switch u := update.(type) {
-	// TODO: this func should accept (xdsclient.EndpointsUpdate, error), and process
-	// the error, instead of having a separate loseContact signal.
-	case xdsclient.EndpointsUpdate:
-		x.edsImpl.handleEDSResponse(u)
-	case *loseContact:
-		// loseContact can be useful for going into fallback.
-	default:
-		panic("unexpected xds client update type")
+func (x *edsBalancer) handleXDSClientUpdate(update *edsUpdate) {
+	if err := update.err; err != nil {
+		// TODO: handle errors from EDS callback.
```

Comment:
Does it make sense to elaborate a little on what actions need to be taken here when we get an error from the callback.

Reply:
Done.